web/src/components/retro-show/show_retro_page.jsx

```diff
@@ -74,6 +74,10 @@ import {retroArchives, retroLogin, retroSettings} from '../../redux/actions/rout
 import {websocketUrl} from '../../helpers/websockets';
 import ArchiveRetroDialog from './archive_retro_dialog';
 import ShareRetroDialog from './share_retro_dialog';
+import {
+  copiedMagicLink,
+  viewedMagicLink,
+} from '../../redux/actions/analytics_actions';
```

Comment:
I don't know redux super well, but are these supposed to be imported into components / crossing this boundary?

Reply:
I spoke to @textbook and he said the pattern used in Posfacto is importing redux actions into page/top level components, to then be passed down and invoked from whoever needs to invoke it. The other analytics example I found was [in the home page](https://github.com/pivotal/postfacto/blob/b018a09132922c0a6d0fc2fda7cb3e14139b7164/web/src/components/home/home_page.jsx#L40).

Reply:
Yes, there are quite a few pages in the app that [`mapDispatchToProps`](https://react-redux.js.org/using-react-redux/connect-mapdispatch) and bring in the action creators this way.

Reply:
Ah, this makes sense now that I see how it's being done elsewhere, thanks!